api/auth/auth.go

```diff
@@ -0,0 +1,154 @@
+package auth
+
+import (
+	"fmt"
+	"os"
+	"os/user"
+	"path/filepath"
+
+	"github.com/jdxcode/netrc"
+	"golang.org/x/crypto/ssh/terminal"
+)
+
+var (
+	// CredentialPath is the path on disk to where credential is recorded
+	CredentialPath string
+	// TTY is the terminal for reading credentials from users
+	TTY *os.File
+)
+
+func init() {
+	// Set tty for normal interactive use
+	TTY = os.Stdin
+
+	// Set CredentialPath for normal interactive use
+	usr, err := user.Current()
+	if err != nil {
+		return
+	}
+	CredentialPath = filepath.Join(usr.HomeDir, ".config", "section", "netrc")
+}
+
+// GetBasicAuth returns credentials for authenticating to the Section API
+func GetBasicAuth() (u, p string, err error) {
+	n, err := netrc.Parse(CredentialPath)
+	if err != nil {
+		return u, p, err
+	}
+	if n.Machine("aperture.section.io") == nil {
+		return u, p, fmt.Errorf("invalid credentials file at %s", CredentialPath)
+	}
+	u = n.Machine("aperture.section.io").Get("login")
+	p = n.Machine("aperture.section.io").Get("password")
+	return u, p, err
+}
+
+// Setup ensures authentication is set up
+func Setup() (err error) {
+	if !IsCredentialRecorded() {
+		Printf(TTY, "No API credentials recorded.\n\n")
+		Printf(TTY, "Let's get you authenticated to the Section API!\n\n")
+
+		m, u, p, err := PromptForCredential()
+		if err != nil {
+			return fmt.Errorf("error when prompting for credential: %s", err)
+		}
+
+		err = WriteCredential(m, u, p)
+		if err != nil {
+			return fmt.Errorf("unable to save credential: %s", err)
+		}
+		return err
+	}
+
+	return err
+}
+
+// IsCredentialRecorded returns if API credentials have been recorded
+func IsCredentialRecorded() bool {
+	n, err := netrc.Parse(CredentialPath)
+	if err != nil {
+		return false
+	}
+	if n.Machine("aperture.section.io") == nil {
+		return false
+	}
+
+	u := n.Machine("aperture.section.io").Get("login")
+	p := n.Machine("aperture.section.io").Get("password")
+	return (len(u) > 0 && len(p) > 0)
+}
+
+// PromptForCredential interactively prompts the user for a credential to authenticate to the Section API
+func PromptForCredential() (m, u, p string, err error) {
+	m = "aperture.section.io"
```

Comment:
Shouldn't all these hard-coded `aperture.section.io` values come instead from the host component of the `SECTION_API_PREFIX`/`--section-api-prefix` overrides, so that credentials for one endpoint aren't sent to another?

Reply:
That's a good point I had not considered.

We should set the machine name in the netrc based on whatever value is set for `api.PrefixURI`.

We should also display what endpoint we're asking for credentials for when showing the prompt. 

I'll make those changes now.

Reply:
Fixed in 3132a77016ed11b7010b4e298a08f6c83a54876b.